### ISSUE TITLE
improvement to summary with fva

### ISSUE THREAD

**Issue by aebrahim** [opencobra/cobrapy#240]
The following improvements would be useful:

 - [x] FVA only runs for reactions which might be displayed (input/output/biomass)
 - [x] solver options should be accepted (and the ability to specify a solver to override the default)
 - [ ] A new optimal flux state is computed (this one can be discussed further)

**Comment by pstjohn** [comment]
* FVA only runs for reactions which might be displayed (input/output/biomass)

This one shouldn't be hard, we'll just move the `.query(lambda x: x, 'boundary')` to above the FVA call. No real point to call FVA on objectives, the range is specified by the function call.

* solver options should be accepted (and the ability to specify a solver to override the default)

Also easy, and currently only relevant to FVA solutions. I'll pass kwargs from the summary call through to the FVA call.

* A new optimal flux state is computed (this one can be discussed further)

So yeah, this one we'd need to think about. They summary methods are designed to print out nicely to a terminal window, so I'll usually call them quickly in series to follow material flow through the network after solving w/ pFBA. Triggering a new optimization at each stage would slow this down, but I agree that as shown in #238 using the existing solution can occasionally lead to unexpected behavior.

**Comment by cdiener** [comment]
I think the third point could be mitigated by redesigning the output format. For instance if the summary output for the FVA case would be something like that:

```bash
PRODUCING REACTIONS -- Pyruvate
-------------------------------
  %             FLUX   RANGE   RXN ID                             REACTION
 40.0%            4        2-8        PYK                  adp_c + h_c + pep_c --> atp_c + pyr_c
 ...
```

Here the percentage and shown flux value will come from the presolved solution and FVA just adds a range. When run without FVA the RANGE column would be missing. This would solve #238 since and would be clear since the presented flux values are "real" (not the mean of the range) and the percentages can be calculated from them.

**Comment by hredestig** [comment]
This all appears fixed with 20d3de8 - recalculating the optimal flux for every call to summary appears heavy and indeed mitigated by new summary output.
